src/test/java/org/javarosa/core/model/TriggerableDagTest.java

```diff
@@ -531,16 +531,237 @@ public void parsing_forms_with_cycles_by_self_reference_in_relevance_should_fail
+                    repeat("/data/group",
+                        input("/data/group/a")
+                    )
+                ),
+                input("/data/b")
+            )
+        ));
+    }
+
+    @Test
+    public void parsing_forms_with_self_reference_cycles_in_fields_of_repeat_groups_should_fail() throws IOException {
+        exceptionRule.expect(RuntimeException.class);
+        exceptionRule.expectMessage("Dependency cycles amongst the xpath expressions in relevant/calculate");
+
+        Scenario.init("Some form", html(
+            head(
+                title("Some form"),
+                model(
+                    mainInstance(t("data id=\"some-form\"",
+                        t("group",
+                            t("a", "1")
+                        )
+                    )),
+                    bind("/data/group/a").type("int").calculate("/data/group[position() = position(current()/..)]/a + 1")
+                )
+            ),
+            body(group("/data/group", repeat("/data/group",
+                input("/data/group/a")
+            )))
+        ));
+    }
+
+    /**
+     * This test fails to parse the form because it thinks there's a self-reference cycle in /data/group/a,
+     * but this would be incorrect because each it depends on the same field belonging to the previous
+     * repeat instance, which wouldn't be a cycle, but an autoincremental feature.
+     */
+    @Test
+    @Ignore
+    public void supports_self_reference_dependency_when_targeting_different_repeat_instance_siblings() throws IOException {
+        Scenario.init("Some form", html(
+            head(
+                title("Some form"),
+                model(
+                    mainInstance(t("data id=\"some-form\"",
+                        t("group",
+                            t("a", "1")
+                        )
+                    )),
+                    bind("/data/group/a").type("int").calculate("/data/group[position() = (position(current()) - 1)]/a + 1")
+                )
+            ),
+            body(group("/data/group", repeat("/data/group",
+                input("/data/group/a")
+            )))
+        ));
+    }
+
+
+    @Test
+    public void parsing_forms_with_cycles_between_fields_of_the_same_repeat_instance_should_fail() throws IOException {
+        exceptionRule.expect(RuntimeException.class);
+        exceptionRule.expectMessage("Dependency cycles amongst the xpath expressions in relevant/calculate");
+        
+        Scenario.init("Some form", html(
+            head(
+                title("Some form"),
+                model(
+                    mainInstance(t("data id=\"some-form\"",
+                        t("group",
+                            t("a", "1"),
+                            t("b", "1")
+                        )
+                    )),
+                    bind("/data/group/a").type("int").calculate("/data/group[position() = position(current())]/b + 1"),
```

Comment:
`/data/group[position() = position(current())]/b` could also be expressed as `../b`, right? I suppose those should have exactly the same behavior so it doesn't matter which is used?